### DISCUSSION TITLE
应该怎么添加自定义代码片段？

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [nshen/learn-neovim-lua#100] [Q&A]
如题！

**Comment by Linya-IronMan** [comment]
小册中只说了设置自定义代码段的选取，跳转。没说怎么设置自定义代码段的映射。
https://juejin.cn/book/7051157342770954277/section/7073710860136349732

**Reply by allworldg** [reply]
https://alpha2phi.medium.com/neovim-for-beginners-snippets-450a855584f8

**Comment by nshen** [comment]
说一下我的，我在写完小册后从 vim-vsnip 转到了 [luasnip](https://github.com/L3MON4D3/LuaSnip)，参考 [friendly-snippets](https://github.com/rafamadriz/friendly-snippets/) 建一个[文件夹](https://github.com/nshen/learn-neovim-lua/tree/v2/lua/cmp/snippets/vscode)里边放自定义补全。

然后在luasnip那边加载这个文件夹

https://github.com/nshen/learn-neovim-lua/blob/19bf2c6b300d7936bdc0f80f38436f906398951d/lua/cmp/luasnip.lua#L7

具体配置在我的v2分支

**Reply by Linya-IronMan** [reply]
V2 的配置很多都不一样了，没办法直接使用。我还特地学了下lua 语言，结果发现还是根本看不懂这些配置。
特别是那些API，应该从哪里查询那些API列表呢？

---

即便是之前版本的配置，实际上我也有很多没有用上。都是使用的你的配置。然而我根本不知道那些配置api对应的含义，尝试一下配置对应的快捷键发现它的行为很奇怪，根本找不到规律，最后只能放弃。

---

请问应该怎么继续深入学习这方面的内容呢？

**Reply by nshen** [reply]
lua语法不是很难，跟js差不多。主要就是投入时间瞎试吧。
插件的API一般就到插件的Github主页和 wiki 页或者文档目录里看。
内置的api 一般 `:h xxx` 或者上 google 搜一搜，我也不是专家，大部分API都不知道是干什么的。

**Reply by allworldg** [reply]
neovim官方文档，所有api都有
